### PR TITLE
[MM-17703] Diff config files

### PR DESCRIPTION
**Pull request by threepwood-mm**
#### Summary

This change adds a make task that allows getting changes in default config files between releases.

To run it, use the following command:
```
make diff-config FROM=5.11.0 TOA=5.13.2
```

This will display the difference between release `5.11.0` and `5.13.2`. 

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/11859

### CODE REVIEW

**Comment by threepwood-mm** [comment]
I'd like to know if the current output (`diff -y`) is enough or there's a different preference.

**Comment by mkraft** [comment]
> I'd like to know if the current output (`diff -y`) is enough or there's a different preference.

@threepwood-mm I'm wondering if it would be useful for the non-devs to include a sample output?

**Comment by threepwood-mm** [comment]
Added a sample of the `diff -y`.

**Comment by jasonblais** [comment]
Thanks @threepwood-mm! I'm curious how changes to default values would be displayed?

**Comment by threepwood-mm** [comment]
> Thanks @threepwood-mm! I'm curious how changes to default values would be displayed?

It shows a `|`, like this:

File: scripts/diff-config.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+if [ -z "$FROM" ]
+then
+  echo "Missing FROM version. Usage: make diff-config FROM=v1.2.3 TO=v1.2.3"
```
Comment (mkraft) [inline scripts/diff-config.sh]:
```suggestion
  echo "Missing FROM version. Usage: make diff-config FROM=5.13.2 TO=5.14.0"
```

Comment (mkraft) [inline scripts/diff-config.sh]:
```suggestion
  echo "Missing FROM version. Usage: make diff-config FROM=5.13.3 TO=5.14.0"
```

File: scripts/diff-config.sh [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  exit 1
+fi
+
+if [ -z "$TO" ]
+then
+  echo "Missing TO version. Usage: make diff-config FROM=v1.2.3 TO=v1.2.3"
```
Comment (mkraft) [inline scripts/diff-config.sh]:
```suggestion
  echo "Missing TO version. Usage: make diff-config FROM=5.13.2 TO=5.14.0"
```

**Comment by threepwood-mm** [comment]
I've added your suggestion, but kept `1.2.3` to make it clearer it's a placeholder. Do you prefer to show a valid usage example instead, like the one you proposed?
Also added the `jq` check, I think it makes a lot of sense.

**Comment by mkraft** [comment]
> I've added your suggestion, but kept `1.2.3` to make it clearer it's a placeholder. Do you prefer to show a valid usage example instead, like the one you proposed?

Yeah, 1.2.3 is cool with me. Only thought is shouldn't the `FROM` and `TO` example values be different?

**Comment by threepwood-mm** [comment]
> > I've added your suggestion, but kept `1.2.3` to make it clearer it's a placeholder. Do you prefer to show a valid usage example instead, like the one you proposed?
> 
> Yeah, 1.2.3 is cool with me. Only thought is shouldn't the `FROM` and `TO` example values be different?

I've no preference on that. For me, `1.2.3` is a placeholder that tells the expected format of the field and the fact that `FROM` and `TO` matches doesn't matter. Is like, for example, if they were integer fields, just showing `0` for both would suffice to make clear what's expecting.
But again, I've no preference, two different placeholder values also make sense to me.

**Comment by mkraft** [comment]
> > > I've added your suggestion, but kept `1.2.3` to make it clearer it's a placeholder. Do you prefer to show a valid usage example instead, like the one you proposed?
> > 
> > 
> > Yeah, 1.2.3 is cool with me. Only thought is shouldn't the `FROM` and `TO` example values be different?
> 
> I've no preference on that. For me, `1.2.3` is a placeholder that tells the expected format of the field and the fact that `FROM` and `TO` matches doesn't matter. Is like, for example, if they were integer fields, just showing `0` for both would suffice to make clear what's expecting.
> But again, I've no preference, two different placeholder values also make sense to me.

My preference would be to have a `TO` value that is ahead of the `FROM` to remove any possible ambiguity.

**Comment by threepwood-mm** [comment]
@grundleborg @amyblais do you have a moment to review this one?

**Comment by hanzei** [comment]
@lindalumitchell Would you please request a QA review for this PR?

**Comment by lindalumitchell** [comment]
Thanks @hanzei, looks good to me to proceed. Removed QA Review label.

**Comment by hanzei** [comment]
Thanks you very much for the contribution @threepwood-mm :tada:

**Pull request merged**